src/register/SearchAbleDropDown.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { Icon } from '@edx/paragon';
+import { ExpandMore, ExpandLess } from '@edx/paragon/icons';
+import onClickOutside from 'react-onclickoutside';
+import PropTypes from 'prop-types';
+import { FormGroup } from '../common-components';
+
+class SearchAbleDropDown extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { value: '', icon: ExpandMore };
+  }
+
+  handleClick = () => {
+    const { options } = this.props;
+
+    if (this.state.value === '') {
+      const dropDownItems = options.map(({ code, name }) => (
+        <button type="button" className="dropdown-item" value={name} key={code} onClick={(e) => { this.countryHandler(e); }}>{name} </button>
+      ));
+      this.setState({ dropDownItems, icon: ExpandLess });
+    }
+  }
+
+    handleOnChange = (e) => {
+      const { options } = this.props;
+      const findstr = e.target.value;
+
+      if (findstr.length >= 3) {
+        let filteredItems = options.filter((option) => (option.name.toLowerCase().includes(findstr.toLowerCase())));
+        filteredItems = filteredItems.map(({ code, name }) => (
+          <button type="button" className="dropdown-item" value={name} key={code} onClick={() => { this.countryHandler(e); }}>{name} </button>
+        ));
+        this.setState({ value: e.target.value, dropDownItems: filteredItems, icon: ExpandLess });
+      } else {
+        this.setState({ value: e.target.value, dropDownItems: '', icon: ExpandMore });
+      }
+    }
+
+    handleClickOutside = (e) => {
+      this.setState(prevState => ({ value: e.target.value || prevState.value, icon: ExpandMore, dropDownItems: '' }));
+    }
+
+    countryHandler(e) {
+      this.setState({ value: e.target.value, dropDownItems: '', icon: ExpandMore });
+    }
+
+    render() {
+      return (
+        <>
+          <FormGroup
+            as="input"
+            name="dropdown"
```

Comment:
We should receive the name in props.
```suggestion
            name="{this.props.name}"
```